// cypress-screenplay/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTask = exports.createQuestion = exports.CypressTask = exports.CypressQuestion = exports.UseCypress = exports.UnsupportedTaskError = exports.AbilityRequestError = exports.Actor = void 0;
var actor_1 = require("./src/actor");
Object.defineProperty(exports, "Actor", { enumerable: true, get: function () { return actor_1.Actor; } });
var errors_1 = require("./src/errors");
Object.defineProperty(exports, "AbilityRequestError", { enumerable: true, get: function () { return errors_1.AbilityRequestError; } });
Object.defineProperty(exports, "UnsupportedTaskError", { enumerable: true, get: function () { return errors_1.UnsupportedTaskError; } });
var cypress_1 = require("./src/cypress");
Object.defineProperty(exports, "UseCypress", { enumerable: true, get: function () { return cypress_1.UseCypress; } });
Object.defineProperty(exports, "CypressQuestion", { enumerable: true, get: function () { return cypress_1.CypressQuestion; } });
Object.defineProperty(exports, "CypressTask", { enumerable: true, get: function () { return cypress_1.CypressTask; } });
Object.defineProperty(exports, "createQuestion", { enumerable: true, get: function () { return cypress_1.createQuestion; } });
Object.defineProperty(exports, "createTask", { enumerable: true, get: function () { return cypress_1.createTask; } });
